efficient zero_mask implementation for vec128_*_neon

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #155766

Differential Revision: [D76481039](https://our.internmc.facebook.com/intern/diff/D76481039/)

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @jerryzh168

swolchok committed:
efficient zero_mask implementation for vec128_*_neon



Differential Revision: [D76481039](https://our.internmc.facebook.com/intern/diff/D76481039/)

[ghstack-poisoned]

swolchok committed:
Update on "efficient zero_mask implementation for vec128_*_neon"


Differential Revision: [D76481039](https://our.internmc.facebook.com/intern/diff/D76481039/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 jerryzh168

[ghstack-poisoned]

swolchok committed:
Update on "efficient zero_mask implementation for vec128_*_neon"


Differential Revision: [D76481039](https://our.internmc.facebook.com/intern/diff/D76481039/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 jerryzh168

[ghstack-poisoned]

malfet reviewed (approved):
If it passes CI, then yeah, I guess it's fine, but would be nice to run some sort of a benchmark to figure out how much efficient that is (and i wonder if later version of compiler might be doing it already)

swolchok commented:
> later version of compiler might be doing it already

-Oz inhibits automatic vectorization

swolchok committed:
Update on "efficient zero_mask implementation for vec128_*_neon"


Differential Revision: [D76481039](https://our.internmc.facebook.com/intern/diff/D76481039/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 jerryzh168

[ghstack-poisoned]

swolchok committed:
Update on "efficient zero_mask implementation for vec128_*_neon"


Differential Revision: [D76481039](https://our.internmc.facebook.com/intern/diff/D76481039/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 jerryzh168

[ghstack-poisoned]

swolchok commented:
@pytorchbot merge
